uat/custom-components/src/main/java/com/aws/greengrass/artifacts/logGenerator.java

```diff
@@ -5,16 +5,141 @@
+    private static double random;
+    private String logFileName;
+    private String logFileExtension;
+    private double fileSize;
+    private String fileSizeUnit;
+    private double logWriteFreqSeconds;
+    private int logMsgSizeBytes;
+    private int fileSizeBytes;
+    private int totalLogNumbers;
+    // public only for unit testing
+    public String currentPath = System.getProperty("user.dir");
+
     @Override
     public void accept(String[] args) {
-        System.out.println("This will be a custom component, now demonstrate we receive some parameters");
-        for (String arg : args) {
-            System.out.println(arg);
+        logFileName = args[0];
+        logFileExtension = args[1];
+        fileSize = Double.parseDouble(args[2]);
+        fileSizeUnit = args[3];
+        logWriteFreqSeconds = Double.parseDouble(args[4]);
+        totalLogNumbers = Integer.parseInt(args[5]);
+
+        fileSizeBytes = getFileSizeInBytes(fileSize, fileSizeUnit);
+
+        // configure the logger by setting the context, rolling policy, encoder
+        Logger logger = configureLogger(currentPath, logFileName, logFileExtension, fileSizeBytes);
+
+        // starting logging message
+        performLogging(logger, totalLogNumbers, logWriteFreqSeconds);
+
+    }
+
+    public static Logger configureLogger(String currentPath, String fileName, String extension, int fileSizeBytes) {
+        LoggerContext loggerContext = new LoggerContext();
+        Logger logger = loggerContext.getLogger("logGenerator");
+
+        // appender: output destination
+        RollingFileAppender<ILoggingEvent> appender = new RollingFileAppender<>();
+        appender.setFile(currentPath + "/" + fileName + "." + extension);
+        appender.setAppend(true);
+        appender.setContext(loggerContext);
+
+        // rolling policy
+        SizeAndTimeBasedRollingPolicy rollingPolicy = new SizeAndTimeBasedRollingPolicy<>();
+        rollingPolicy.setMaxFileSize(new FileSize(fileSizeBytes));
+        rollingPolicy.setMaxHistory(MAX_HISTORY);
+        rollingPolicy.setFileNamePattern(currentPath + "/" + fileName + rotationNamePattern + "." + extension);
+        rollingPolicy.setParent(appender);
+        rollingPolicy.setContext(loggerContext);
+        appender.setRollingPolicy(rollingPolicy);
+        rollingPolicy.start();
+
+        // encoder
+        PatternLayoutEncoder encoder = new PatternLayoutEncoder();
+        encoder.setPattern("%d{yyyy-MM-dd HH:mm:ss} %-5level %logger{36} - %msg%n");
+        encoder.setContext(loggerContext);
+        encoder.start();
+
+        // attach encoder to appender, then start
+        appender.setEncoder(encoder);
+        appender.start();
+
+        // set logger
+        logger.addAppender(appender);
+        logger.setLevel(Level.INFO);
+
+        return logger;
+    }
+
+    public static void performLogging(Logger logger, int totalLogNumbers, double logWriteFreqSeconds) {
```

Comment:
Why is this public?. If you really want to test it then make this private and the call `accept`. Test things from a consumers perspective, what the caller can actually do. That will make your tests impervious to changes allowing you to freely change your component. They will only change if you change the ordering of the params you use to call `accept`